cmd/federation-controller/main.go

```diff
@@ -5,35 +5,62 @@ import (
 	"encoding/json"
 	"flag"
 	"fmt"
-	discovery "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v3"
-	"github.com/jewertow/federation/internal/pkg/federation"
-	"github.com/jewertow/federation/internal/pkg/xds/adsc"
-	"log"
 	"os"
 	"os/signal"
 	"sync"
 	"syscall"
 
+	discovery "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v3"
+	"github.com/jewertow/federation/internal/pkg/federation"
+	"github.com/jewertow/federation/internal/pkg/xds/adsc"
+
 	"github.com/jewertow/federation/internal/pkg/config"
 	"github.com/jewertow/federation/internal/pkg/mcp"
 	"github.com/jewertow/federation/internal/pkg/xds"
 	"github.com/jewertow/federation/internal/pkg/xds/adss"
+	"github.com/spf13/cobra"
+	"istio.io/istio/pkg/log"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/client-go/informers"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
-	"k8s.io/klog/v2"
 )
 
-// Global variable to store the parsed arguments and "flag" arguments
+// Global variable to store the parsed commandline arguments.
 var (
-	meshPeer = flag.String("meshPeers", "",
+	meshPeer, exportedServiceSet, importedServiceSet *string
```

Comment:
```suggestion
	meshPeer, exportedServiceSet, importedServiceSet string
```
These pointers are uninitialized, so I am getting nil pointer dereference error when I execute federation-controller:
```
make
./out/federation-controller --meshPeers '{}' --exportedServiceSet '{}'
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x1840664]

goroutine 1 [running]:
github.com/spf13/pflag.newStringValue(...)
        /home/jewertow/go/pkg/mod/github.com/spf13/pflag@v1.0.5/string.go:7
github.com/spf13/pflag.(*FlagSet).StringVar(...)
        /home/jewertow/go/pkg/mod/github.com/spf13/pflag@v1.0.5/string.go:37
main.addFlags(0xc0002c0008)
        /home/jewertow/oss/federation/cmd/federation-controller/main.go:54 +0x24
main.NewRootCommand()
        /home/jewertow/oss/federation/cmd/federation-controller/main.go:48 +0x71
main.main()
        /home/jewertow/oss/federation/cmd/federation-controller/main.go:123 +0x34
```

Reply:
Fixed by following the suggested change. 
I will add some e2e main_tests to cover this case in the next commit.